ocaml/lib/snarky_bindings.ml

```diff
@@ -236,6 +236,24 @@ module Poseidon = struct
         Poseidon_sponge.squeeze s |> Impl.Field.constant
 end
 
+module SHA = struct
+  let create message nist length =
+    let message_array = Array.to_list message in
+    if Js.to_bool nist then
+      Kimchi_gadgets.Keccak.nist_sha3 (module Impl) length message_array
+    else Kimchi_gadgets.Keccak.ethereum (module Impl) message_array
+
+  let field_bytes_of_hex hex =
+    Array.of_list
+      (Kimchi_gadgets.Common.field_bytes_of_hex
+         (module Impl)
+         (Js.to_string hex) )
+
+  let check_bits value bits =
+    Kimchi_gadgets.Lookup.less_than_bits (module Impl) ~bits value ;
+    Kimchi_gadgets.Range_check.bits64 (module Impl) Impl.Field.zero
```

Comment:
The reason for calling `Kimchi_gadgets.Range_check.bits64` is because `less_than_bits` expects a "dummy gate" to be initialized before calling it (as noted [here](https://github.com/MinaProtocol/mina/blob/bd1688a5861ff4431dbaab06e0d39129d259c307/src/lib/crypto/kimchi_backend/gadgets/lookup.ml#L31)). Calling this every time `check_bits` is called seems wasteful, is there a better way to initialize the range check once?